pkg/inst-api-semconv/instrumenter/http/http_attrs_extractor.go

```diff
@@ -0,0 +1,118 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package http
+
+import (
+	"context"
+	"github.com/open-telemetry/opentelemetry-go-compile-instrumentation/pkg/inst-api-semconv/instrumenter/utils"
+	"go.opentelemetry.io/otel/attribute"
+	semconv "go.opentelemetry.io/otel/semconv/v1.30.0"
+	"sync/atomic"
+)
+
+type HttpRequest interface{}
+
+type HttpResponse interface{}
+
+type HttpCommonAttrsExtractor[REQUEST HttpRequest, RESPONSE HttpResponse, GETTER1 HttpCommonAttrsGetter[REQUEST, RESPONSE]] struct {
```

Comment:
This is too long for a single line, should we have a spec to have fix length for a single line?

Reply:
I've add limited the max columns in my local IDE, I will add the limitation in golang-lint check in a new PR